source/main.c

```diff
@@ -256,31 +357,65 @@ int main()
     if (load_fat("sd2", &__io_gcsd2)) goto load;
 
 load:
-    // Wait to exit while the d-pad down direction is held.
-    while (all_buttons_held & PAD_BUTTON_DOWN)
-    {
-        VIDEO_WaitVSync();
-        PAD_ScanPads();
-        all_buttons_held = (
-            PAD_ButtonsHeld(PAD_CHAN0) |
-            PAD_ButtonsHeld(PAD_CHAN1) |
-            PAD_ButtonsHeld(PAD_CHAN2) |
-            PAD_ButtonsHeld(PAD_CHAN3)
-        );
-    }
-
     if (dol)
     {
+        struct __argv dolargs;
+        dolargs.commandLine = (char *) NULL;
+        dolargs.length = 0;
+        
+        // https://github.com/emukidid/swiss-gc/blob/f5319aab248287c847cb9468325ebcf54c993fb1/cube/swiss/source/aram/sidestep.c#L350
+        if (dol_argc)
+        {
+            dolargs.argvMagic = ARGV_MAGIC;
+            dolargs.argc = dol_argc;
+            dolargs.length = 1;
+
+            for (int i = 0; i < dol_argc; i++)
+            {
+                size_t arg_length = strlen(dol_argv[i]) + 1;
+                dolargs.length += arg_length;
+            }
+
+            kprintf("CLI argv size is %iB\n", dolargs.length);
+            dolargs.commandLine = (char *) malloc(dolargs.length);
+
+            if (!dolargs.commandLine)
+            {
+                kprintf("Couldn't allocate memory for CLI argv\n");
+                dolargs.length = 0;
+            }
+            else
+            {
+                unsigned int position = 0;
+                for (int i = 0; i < dol_argc; i++)
+                {
+                    size_t arg_length = strlen(dol_argv[i]) + 1;
+                    memcpy(dolargs.commandLine + position, dol_argv[i], arg_length);
+                    position += arg_length;
+                }
+                dolargs.commandLine[dolargs.length - 1] = '\0';
+                DCStoreRange(dolargs.commandLine, dolargs.length);
+                
+                // kprintf("argc: %i\n", dol_argc);
+                // for (int i = 0; i < dol_argc; ++i) {
+                //     kprintf("arg%i: %s\n", i, dol_argv[i]);
+                // }
```

Comment:
You can leave this in for verbose debugging, just change the comments to `#if 0`/`#endif`, that way it's easy to enable.